sqlitecluster/SQLiteSequentialNotifier.cpp

```diff
@@ -46,30 +50,33 @@ void SQLiteSequentialNotifier::notifyThrough(uint64_t value) {
     if (value > _value) {
         _value = value;
     }
-    auto lastToDelete = _valueToPendingThreadMap.begin();
-    for (auto it = _valueToPendingThreadMap.begin(); it != _valueToPendingThreadMap.end(); it++) {
-        if (it->first > value)  {
-            // If we've passed our value, there's nothing else to erase, so we can stop.
-            break;
-        }
+    for (auto valueThreadMapPtr : {&_valueToPendingThreadMap, &_valueToPendingThreadMapNoCurrentTransaction}) {
+        auto& valueThreadMap = *valueThreadMapPtr;
+        auto lastToDelete = valueThreadMap.begin();
+        for (auto it = valueThreadMap.begin(); it != valueThreadMap.end(); it++) {
+            if (it->first > value)  {
+                // If we've passed our value, there's nothing else to erase, so we can stop.
+                break;
+            }
 
-        // Note that we'll delete this item from the map.
-        lastToDelete++;
+            // Note that we'll delete this item from the map.
+            lastToDelete++;
 
-        // Make the changes to the state object - mark it complete and notify anyone waiting.
-        lock_guard<mutex> lock(it->second->waitingThreadMutex);
-        it->second->result = RESULT::COMPLETED;
-        it->second->waitingThreadConditionVariable.notify_all();
-    }
+            // Make the changes to the state object - mark it complete and notify anyone waiting.
+            lock_guard<mutex> lock(it->second->waitingThreadMutex);
+            it->second->result = RESULT::COMPLETED;
+            it->second->waitingThreadConditionVariable.notify_all();
```

Comment:
This one should be fast because it only notifies waiting threads for numbers through `value` which should just be the first couple in the list. If it's more than that, there's not much we can do to make it faster since they all need to receive the notification anyway.